### ISSUE TITLE
Integration tests take a considerable amount of time, mostly due to uploads

### ISSUE THREAD

**Issue by fearful-symmetry** [elastic/elastic-agent#2842]
Right now, for me, `mage integration:test` takes about an hour, and a considerable amount of that time is spent uploading build artifacts to GCP. I did a test with `nethogs`, and during a test mage will upload 1907.854 MBs. I have pretty decent internet, but for me this upload step takes a little more than half of the total runtime. Unless we intend people to run tests on other GCP instances that might have faster upload speeds, we should look for a way to optimize this, and the tests in general. As a holdover, we might want to add some build constraints (if they don't exist already) that limit the amount of uploading that `mage integration:test` will do.

**Comment by blakerouse** [comment]
`mage integration:test` is already optimized to only upload the build of the Elastic Agent that is required for that OS and architecture combination, it does **NOT** upload all builds to each runner.

Have you tried to adjust the region in google cloud that would provide you better performance? You can set `TEST_INTEG_AUTH_GCP_DATACENTER` to adjust the region. If you change the region it would also be good to make that region match for ESS with the `TEST_INTEG_AUTH_ESS_REGION` environment variables.

**Comment by fearful-symmetry** [comment]
@blakerouse are those env vars documented? If not, we should probably add add them.

**Comment by fearful-symmetry** [comment]
Well, it appears that fix has other issues:
```
 Error: error running test: failed to create ESS cloud for version 8.9.0-SNAPSHOT: failed to create: (resources.region_not_allowed) The regions [gcp-us-west1] are not allowed
```

**Comment by blakerouse** [comment]
@fearful-symmetry Glad you got better performance. We will hit a performance limit on the amount of time that it takes always. At some point the time it takes to provision the machines, plus installation of required packages, setup of golang and mage cannot be avoided.

Windows takes even longer, and that is mainly waiting on Windows to deploy just to be ready to actually perform work on it.

**Comment by fearful-symmetry** [comment]
Going to file a separate issue on optimizations.

**Comment by jlind23** [comment]
@fearful-symmetry now that you created another issue, should we close this one?

**Comment by fearful-symmetry** [comment]
Yah, we can continue in https://github.com/elastic/elastic-agent/issues/2856
